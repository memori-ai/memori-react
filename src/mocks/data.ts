import {
  Medium,
  Memori,
  Message,
  Memory,
  Integration,
  Tenant,
  DialogState,
  KnownFact,
  ExpertReference,
  User,
  Venue,
} from '@memori.ai/memori-api-client/dist/types';

export const sessionID = '131165be-9d1a-42fb-a3ce-e8f86d40c88f';

export const tenant: Tenant = {
  adminCount: 3,
  config: {
    name: 'AIsuru',
    requirePosition: false,
    showNewUser: true,
  },
  creationTimestamp: '2023-05-31T14:32:48.885287Z',
  description: 'AIsuru',
  disableRegistration: false,
  id: 'www.aisuru.com',
  lastChangeTimestamp: '2023-05-31T14:32:48.885287Z',
  logoURL: 'https://aisuru.com/images/aisuru/logo.png',
  maxAdmins: 0,
  maxCompletions: 0,
  maxCompletionsPerUser: 0,
  maxFreeSessions: 400,
  maxFreeSessionsPerUser: 100,
  maxMemoriPerAdmin: 0,
  maxMemoriPerUser: 3,
  maxTotalMemori: 0,
  maxUsers: 0,
  memoriCount: 0,
  name: 'www.aisuru.com',
  nonFreeSessionCost: 0.02,
  paying: true,
  tenantID: '96caa4b4-31a4-48e5-8163-dec61869a2a7',
  theme: 'aisuru',
  userCount: 0,
};

export const user: User = {
  tenant: 'localhost:3000',
  userID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  userName: 'nzambello',
  eMail: 'nicola@nzambello.dev',
  admin: false,
  maxMemori: 0,
  enableMemoriCreation: true,
  enableBoardOfExperts: true,
  maxFreeSessions: 0,
  tnCAndPPAccepted: true,
  tnCAndPPAcceptanceDate: '2021-03-01T00:00:00.000Z',
  pAndCUAccepted: true,
  pAndCUAcceptanceDate: '2021-03-01T00:00:00.000Z',
  birthDate: '1900-03-01T00:00:00.000Z',
  age: 28,
  avatarURL: 'https://avatars.githubusercontent.com/u/21101435?v=4',
};

export const memori: Memori = {
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  engineMemoriID: '66b4e161-2431-4b21-9b70-d8c27de730ca',
  name: 'Memori',
  memoriConfigurationID: 'MemoriCloud-it_IT',
  description: 'Lorem ipsum.',
  voiceType: 'male',
  isGiver: true,
  isReceiver: false,
  privacyType: 'PUBLIC',
  needsPosition: false,
  culture: 'it-IT',
  categories: [],
  publishedInTheMetaverse: true,
  exposed: true,
  enableCompletions: true,
  nsfw: false,
  ageRestriction: 14,
  contentQualityIndex: 66.6,
  contentQualityIndexTimestamp: '2021-03-01T12:00:00.000Z',
  ownerUserName: 'username',
  ownerTenantName: 'aisuru.com',
  ownerUserID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  metaverseEnvironment: 'synthwave',
  secretToken: 'awanagana',
  giverPIN: '1234',
  giverTag: 'üßë‚Äçüíª',
  avatarURL: 'https://aisuru.com/images/aisuru/square_logo.png',
  coverURL: 'https://aisuru.com/images/aisuru/og-image.png',
};

export const history: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const historyWithMedia: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui delle cose per te.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai/en',
        mimeType: 'text/html',
        title: 'Link Memori Srl',
      },
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efce',
        url: 'https://rawmaterial.it/en',
        mimeType: 'text/html',
        title: 'Link RawMaterial',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
        mimeType: 'text/html',
        title: 'Introducing Plone Remix | Vimeo',
        url: 'https://vimeo.com/766468314',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5969',
        mimeType: 'text/html',
        title: 'A sustainable web: is it possible? - Nicola Zambello | YouTube',
        url: 'https://www.youtube.com/watch?v=feH26j3rBz8',
      },
      ...Array.from({ length: 3 }, (_, i) => ({
        mediumID: `95226d7e-7bae-465e-8b80-995587bb597${i}`,
        mimeType: 'image/png',
        title: `Image ${i}`,
        url: `https://picsum.photos/${i % 2 ? '200' : '300'}/${
          i % 3 ? '300' : '200'
        }?random=${i}`,
      })),
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
];

export const historyWithAIGeneratedMessages: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Sa. Sa. Prova',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    generatedByAI: true,
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const dialogState: DialogState = {
  state: 'R1',
  previousState: 'I0',
  stateName: 'WaitingForReceiverQuestion',
  confidence: 1,
  knownTags: {
    '‚ò†Ô∏è': 'test',
    'üòé': 'Ciccio',
  },
  emission: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
  hints: ['Va bene', 'No grazie'],
  media: [],
  acceptsTimeout: true,
  acceptsAbort: false,
  acceptsMedia: false,
  acceptsDate: false,
  acceptsPlace: false,
  acceptsTag: false,
  giverID: 'c832e2dc-403c-4baf-a3b7-2374e100dbcf',
  contextVars: {},
};

export const medium: Medium = {
  mediumID: 'a196b513-d745-4121-8913-8f457b999add',
  url: 'https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2',
  mimeType: 'image/jpeg',
  title: 'Game Cover',
  creationTimestamp: '2022-05-03T08:57:30.584439Z',
  creationName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
  lastChangeTimestamp: '2022-05-03T14:19:09.932865Z',
  lastChangeName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
};

export const integration: Integration = {
  integrationID: 'cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1',
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  type: 'LANDING_EXPERIENCE',
  state: 'NEW',
  deviceEmails: [],
  customData:
    '{"textColor":"#2a2a2a","buttonBgColor":"#823ce1","buttonTextColor":"#ffffff","name":"Web","globalBackground":"https://assets.memori.ai/api/v2/asset/364e498c-11da-42d5-9e32-19e5d137d4b8.jpeg","blurBackground":true,"innerBgColor":"light","innerBgAlpha":0.8,"multilanguage":true,"avatar":"readyplayerme","avatarURL":"https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb#1669663599444"}',
  resources: [],
  publish: true,
  creationTimestamp: '2022-06-11T14:13:45.685038Z',
  lastChangeTimestamp: '2022-06-11T14:13:45.685038Z',
};

export const knownFact: KnownFact = {
  knownFactID: 'b0b0b0b0-b0b0-b0b0-b0b0-b0b0b0b0b0b3',
  knownFactType: 'ShortTerm',
  text: 'I am a known fact. Quisque in ultrices lectus. Nulla at urna diam. Proin sodales lobortis libero eu facilisis.',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const expertReference: ExpertReference = {
  expertID: 'f016f204-c307-483b-8891-680a3c974c53',
  name: 'TEST MEMORI',
  description: 'lui √® competente',
  default: true,
  expertMemoriID: 'dfc44f1c-1ba7-4e1e-a234-28aa8b6b3d32',
  expertBaseURL: 'http://localhost:7778',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const venue: Venue = {
  latitude: 44.66579,
  longitude: 11.48823,
  placeName: 'Altedo, Bologna, Italy',
  uncertainty: 2,
};

export const memoryQuestion: Memory = {
  memoryID: '08c0697d-f7f3-4a47-9970-aff75f01fb6c',
  memoryType: 'Question',
  lastRead: undefined,
  readOccurrences: undefined,
  receiverID: undefined,
  receiverTag: undefined,
  receiverName: undefined,
  media: [
    {
      mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
      mimeType: 'text/html',
      url: 'https://rawmaterial.it/it',
      title: 'RawMaterial',
    },
  ],
  title: 'sei umano',
  titleVariants: ['sei reale'],
  answers: [
    {
      text: 'se ti dico si ci credi?',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:25.410377',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:25.410406',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
    {
      text: 'probabilmente lo sono pi√π di te',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:36.957656',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:36.957657',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
  ],
  date: undefined,
  dateUncertaintyDays: undefined,
  placeName: undefined,
  placeLatitude: undefined,
  placeLongitude: undefined,
  placeUncertaintyKm: undefined,
  preformatted: false,
  conclusive: false,
  notPickable: false,
  contextVarsToSet: {
    SUBJECT: 'TRAVEL',
  },
  contextVarsToMatch: {
    SUBJECT: 'TRANSPORTS',
  },
  creationTimestamp: '2022-03-23T09:37:16.568149',
  creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
  lastChangeTimestamp: '2022-03-23T09:37:36.957695',
  lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
};
