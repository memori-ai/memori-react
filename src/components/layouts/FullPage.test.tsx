import { render } from '@testing-library/react';
import Memori from '../MemoriWidget/MemoriWidget';
import { integration, memori, tenant } from '../../mocks/data';

it('renders FullPage layout unchanged', () => {
  const { container } = render(
    <Memori
      showShare={true}
      showSettings={true}
      memori={memori}
      tenant={tenant}
      integration={integration}
      layout="FULLPAGE"
    />
  );
  expect(container).toMatchSnapshot();
});
