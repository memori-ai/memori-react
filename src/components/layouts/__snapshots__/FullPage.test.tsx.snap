// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FullPage layout unchanged 1`] = `
<div>
  <div
    class="memori memori-widget memori-layout-fullpage memori-controls-bottom memori--avatar-readyplayerme memori--with-integration"
    data-memori-engine-state="{}"
    data-memori-id="66b4e161-2431-4b21-9b70-d8c27de730ca"
    data-memori-integration="cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1"
    data-memori-name="Memori"
    data-memori-secondary-id="25ced51c-3520-41af-8bbe-222d861b8e32"
    style="height: 100vh;"
  >
    <style>
      
    :root, .memori-widget, .memori-drawer, .memori-modal {
      --memori-chat-bubble-bg: #fff;
--memori-text-color: #2a2a2a;
--memori-button-bg: #823ce1;
--memori-primary: #823ce1;
--memori-button-text: #ffffff;
--memori-blur-background: 5px;
--memori-inner-bg: rgba(255, 255, 255, 0.8);
--memori-inner-content-pad: 1.5rem;
--memori-nav-bg-image: none;
--memori-nav-bg: rgba(255, 255, 255, 0.8);
    }
  
    </style>
    <div
      class="memori--global-background"
    >
      <div
        class="memori--global-background-image"
        style="background-image: url(https://assets.memori.ai/api/v2/asset/364e498c-11da-42d5-9e32-19e5d137d4b8.jpeg);"
      />
    </div>
    <div
      class="memori-spin"
    >
      <div
        class="memori-header"
      >
        <div
          class="memori-export-history-button memori-header--button memori-header--button--export"
          data-headlessui-state=""
          title="Vuoi esportare la cronologia chat?"
        >
          <button
            class="memori-button memori-button--circle memori-button--icon-only memori-button--disabled"
            data-headlessui-state=""
            disabled=""
            id="headlessui-popover-button-:r0:"
            type="button"
          >
            <div
              class="memori-button--icon"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                role="img"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                />
              </svg>
            </div>
          </button>
        </div>
        <button
          class="memori-button memori-button--primary memori-button--circle memori-button--padded memori-button--icon-only memori-header--button memori-header--button-settings"
          title="Impostazioni"
        >
          <span
            class="memori-button--icon"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              role="img"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
              />
            </svg>
          </span>
        </button>
        <div
          class="memori-share-button memori-share-button--align-left"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="memori-button memori-button--circle memori-button--icon-only memori-share-button--button memori-header--button memori-header--button-share memori-button--primary"
            data-headlessui-state=""
            id="headlessui-menu-button-:r5:"
            title="Condividi"
            type="button"
          >
            <div
              class="memori-button--icon"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                role="img"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 0 0 0-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        class="memori--grid"
      >
        <div
          class="memori--grid-column memori--grid-column-left"
        >
          <div
            class="memori--avatar-wrapper memori--avatar-readyplayerme"
          >
            <div
              class="memori--blob-container"
            >
              <div
                class="memori-blob"
              >
                <div
                  class="mainDiv"
                />
                <div
                  class="mainDiv"
                />
                <div
                  class="mainDiv"
                />
                <div
                  class="mainDiv"
                />
                <div
                  class="mainDiv"
                />
              </div>
            </div>
          </div>
          <div
            class="memori--avatar-toggle"
          >
            <button
              class="memori-button memori-button--ghost memori-button--rounded memori-button--padded memori-button--with-icon"
            >
              <span
                class="memori-button--icon"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  role="img"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                  />
                  <path
                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                  />
                </svg>
              </span>
              <span
                class="memori--avatar-toggle-text"
              >
                Nascondi
              </span>
            </button>
          </div>
          <div
            id="extension"
          />
        </div>
        <div
          class="memori--grid-column memori--grid-column-right"
        >
          <div
            class="memori--start-panel"
          >
            <div
              class="memori--cover"
            >
              <div
                class="memori--completions-enabled"
              >
                <div
                  class="memori-tooltip memori-tooltip--align-left"
                >
                  <div
                    class="memori-tooltip--content"
                  >
                    IA evoluta, può rispondere con risposte generate automaticamente che talvolta potrebbero contenere informazioni non corrette
                  </div>
                  <div
                    class="memori-tooltip--trigger"
                  >
                    <span
                      aria-label="IA evoluta, può rispondere con risposte generate automaticamente che talvolta potrebbero contenere informazioni non corrette"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          clip-rule="evenodd"
                          fill="currentColor"
                          fill-rule="evenodd"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M0 4a4 4 0 014-4h16a4 4 0 014 4v16a4 4 0 01-4 4H4a4 4 0 01-4-4zm4-2.4A2.4 2.4 0 001.6 4v16A2.4 2.4 0 004 22.4h16a2.4 2.4 0 002.4-2.4V4A2.4 2.4 0 0020 1.6z"
                            fill-rule="evenodd"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M9.715 8.442a.798.798 0 00-1.43 0l-3.2 6.4a.799.799 0 101.431.716l.579-1.158h3.811l.578 1.158a.8.8 0 001.431-.716zm.391 4.358L9 10.589 7.894 12.8z"
                            fill-rule="evenodd"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M17 8c.552 0 1 .358 1 .8v6.4c0 .442-.448.8-1 .8s-1-.358-1-.8V8.8c0-.442.448-.8 1-.8z"
                            fill-rule="evenodd"
                          />
                        </g>
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <picture
              class="memori--avatar"
            >
              <source
                src="https://aisuru.com/images/aisuru/square_logo.png"
              />
              <img
                alt="Memori"
                src="https://aisuru.com/images/aisuru/square_logo.png"
              />
            </picture>
            <h2
              class="memori--title"
            >
              Memori
            </h2>
            <div
              class="memori--description"
            >
              <p>
                <span
                  class="memori--description-text"
                >
                  Lorem ipsum.
                </span>
              </p>
              <div
                class="memori--language-chooser"
              >
                <label
                  for="user-lang-pref"
                  id="user-lang-pref-label"
                >
                  Voglio parlare con Memori in
                </label>
                <select
                  aria-labelledby="user-lang-pref-label"
                  class="memori-select--button"
                  id="user-lang-pref"
                >
                  <option
                    aria-label="Italiano"
                    value="IT"
                  >
                    Italiano
                  </option>
                  <option
                    aria-label="English"
                    value="EN"
                  >
                    English
                  </option>
                  <option
                    aria-label="Deutsch"
                    value="DE"
                  >
                    Deutsch
                  </option>
                  <option
                    aria-label="Español"
                    value="ES"
                  >
                    Español
                  </option>
                  <option
                    aria-label="Français"
                    value="FR"
                  >
                    Français
                  </option>
                  <option
                    aria-label="Português"
                    value="PT"
                  >
                    Português
                  </option>
                  <option
                    aria-label="Українська"
                    value="UK"
                  >
                    Українська
                  </option>
                  <option
                    aria-label="Русский"
                    value="RU"
                  >
                    Русский
                  </option>
                  <option
                    aria-label="Polski"
                    value="PL"
                  >
                    Polski
                  </option>
                  <option
                    aria-label="Suomi"
                    value="FI"
                  >
                    Suomi
                  </option>
                  <option
                    aria-label="Ελληνικά"
                    value="EL"
                  >
                    Ελληνικά
                  </option>
                  <option
                    aria-label="العربية"
                    value="AR"
                  >
                    العربية
                  </option>
                  <option
                    aria-label="中文"
                    value="ZH"
                  >
                    中文
                  </option>
                  <option
                    aria-label="日本語"
                    value="JA"
                  >
                    日本語
                  </option>
                </select>
              </div>
              <button
                class="memori-button memori-button--primary memori-button--rounded memori-button--padded memori--start-button"
              >
                INIZIAMO
              </button>
              <p
                class="memori--start-description"
              >
                Per parlare con me clicca sul pulsante INIZIAMO per cominciare
              </p>
            </div>
          </div>
        </div>
        <div
          class="memori--powered-by"
        >
          <img
            alt=""
            src="https://aisuru.com/images/aisuru/logo.png"
          />
          <p>
            <span
              class="sr-only"
            >
              Powered by
            </span>
            <a
              href="https://memori.ai/it?integrationID=cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1&memori=aisuru.com-username-Memori"
              rel="noopener noreferrer"
              target="_blank"
            >
              Memori.AI
            </a>
          </p>
        </div>
      </div>
      <div
        class="memori-spin--spinner"
      >
        <svg
          aria-hidden="true"
          class="memori-loading-icon"
          focusable="false"
          role="img"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </div>
    </div>
    <audio
      id="memori-audio"
      src="https://aisuru.com/intro.mp3"
      style="display: none;"
    />
  </div>
  <script
    async=""
    id="MathJax-script"
    src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"
  />
</div>
`;
